ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -213,11 +215,16 @@ private TableRowsEntity handlePullQuery(
       final ServiceContext serviceContext,
       final PullQueryContext pullQueryContext
   ) {
+    final KsqlConfig ksqlConfigWithOverrides = statement.getConfig()
```

Comment:
Is the cloning happening per pull query ? if so, wont it be expensive?

Reply:
It is.  To be fair, for select statements that aren't select *, it's happening anyway, if you look below in this file, so I just factored it up a bit so it always happens.

It copies over explicit overrides as well as originals which are the fields set in the config (though not default values, I don't believe).  I was largely copying the existing mechanism for doing this sort of thing, though it may not be the best.  I could try to do a simplified version of the check.  What do you think?

Reply:
It seems that the configs we get from the overrides are never used outside of here. Moreover, we only care about a single one, the max lag.  An alternative idea is: Don't override the default max lag in `KsqlConfig` since either way the value is so short lived, per pull query. Rather just parse the value provided in the statement into a local variable. If no value is provided, then use the default one from `KsqlConfig`. But don't do any updates on `KsqlConfig`.